src/Adapters/Push/SNSPushAdapter.js

```diff
@@ -0,0 +1,186 @@
+"use strict";
+// SNSAdapter
+//
+// Uses SNS for push notification
+import PushAdapter from './PushAdapter';
+
+const Parse = require('parse/node').Parse;
+const GCM = require('../../GCM');
+
+const AWS = require('aws-sdk');
+
+var DEFAULT_REGION = "us-east-1";
+import { classifyInstallations } from './PushAdapterUtils';
+
+export class SNSPushAdapter extends PushAdapter {
+
+    // Publish to an SNS endpoint
+    // Providing AWS access and secret keys is mandatory
+    // Region will use sane defaults if omitted
+    constructor(pushConfig = {}) {
+        super(pushConfig);
+        this.validPushTypes = ['ios', 'android'];
+        this.availablePushTypes = [];
+        this.arnMap = {};
+        this.senderMap = {};
+
+        if (!pushConfig.accessKey || !pushConfig.secretKey) {
+            throw new Parse.Error(Parse.Error.PUSH_MISCONFIGURED,
+                'Need to provide AWS keys');
+        }
+
+        if (pushConfig.pushTypes) {
+            let pushTypes = Object.keys(pushConfig.pushTypes);
+            for (let pushType of pushTypes) {
+                if (this.validPushTypes.indexOf(pushType) < 0) {
+                    throw new Parse.Error(Parse.Error.PUSH_MISCONFIGURED,
+                        'Push to ' + pushTypes + ' is not supported');
+                }
+                this.availablePushTypes.push(pushType);
+                switch (pushType) {
+                    case 'ios':
+                        this.senderMap[pushType] = this.sendToAPNS.bind(this);
+                        this.arnMap[pushType] = pushConfig.pushTypes[pushType];
+                        break;
+                    case 'android':
+                        this.senderMap[pushType] = this.sendToGCM.bind(this);
+                        this.arnMap[pushType] = pushConfig.pushTypes[pushType];
+                        break;
+                }
+            }
+        }
+
+        AWS.config.update({
+            accessKeyId: pushConfig.accessKey,
+            secretAccessKey: pushConfig.secretKey,
+            region: pushConfig.region || DEFAULT_REGION
+        });
+
+        // Instantiate after config is setup.
+        this.sns = new AWS.SNS();
+    }
+
+    getValidPushTypes() {
+        return this.availablePushTypes;
+    }
+
+    static classifyInstallations(installations, validTypes) {
+        return classifyInstallations(installations, validTypes)
+    }
+
+    //Generate proper json for APNS message
+    static generateiOSPayload(data) {
+        return {
+            'APNS': JSON.stringify(data)

```

Comment:
Yeah I ended up just importing from the APNS module to generate the payload now...